libr/core/cmd_debug.c

```diff
@@ -792,6 +795,169 @@ static void cmd_debug_modules(RCore *core, int mode) { // "dmm"
 	r_list_free (list);
 }
 
+static int cmd_debug_map_heap(RCore *core, const char *input) {
+
+	const char* help_msg[] = {
+		"Usage:", "dmh", " # Memory map heap",
+		"dmha", "", "Struct Malloc State (main_arena)",
+		"dmh?", "", "",
+		NULL
+	};
+
+	RListIter *iter;
+	RDebugMap *map;
+	
+	switch (input[0]) {
+	
+	case 'a': // "dmha"
+		{
+		const char *dir_dbg = "/usr/lib/debug";
+		const char *dir_dbg_build_id = "/usr/lib/debug/.build-id";
+		const char *symname = "main_arena";
+		const char *libc_ver = NULL;
+		const char *libc_ver_end = NULL;
+		ut64 main_arena = 0LL;
+		ut64 libc_addr = 0LL;
+
+		r_debug_map_sync (core->dbg); 
+
+		r_list_foreach (core->dbg->maps, iter, map) {
+			if (strstr (map->name, "/libc-")) {
+				libc_addr = (NBYTES == 4) ? (map->addr_end) : (map->addr);
+				libc_ver_end = map->name;
+				break;
+			}
+		}
+
+		if (access (dir_dbg, F_OK) == -1) {
+			eprintf ("Debug directory not found at /usr/lib, is libc<version>-dbg installed?\n");
+			break;			
+		}
+
+		if (access (dir_dbg_build_id, F_OK) == -1) {
+			libc_ver = dir_dbg;
+		} else {
+			RListIter *iter;
```

Comment:
All this code should be on his own function or something to isolate as long as it will work only with a glibc version